python/ribasim/ribasim/model.py

```diff
@@ -232,14 +232,15 @@ def _write_toml(self, fn: FilePath):
         return fn
 
     def _save(self, directory: DirectoryPath, input_dir: DirectoryPath):
+        self.network._save(directory, input_dir)
         for sub in self.nodes().values():
             sub._save(directory, input_dir)
 
     def nodes(self):
         return {
             k: getattr(self, k)
             for k in self.model_fields.keys()
-            if isinstance(getattr(self, k), NodeModel)
+            if isinstance(getattr(self, k), NodeModel) and k != "network"
         }
```

Comment:
I left out the network since it really isn't a node, and is such a different beast that it usually needs to be handled separately from the nodes.